src/NuGet.Clients/NuGet.Console/OutputConsole/ChannelOutputConsole.cs

```diff
@@ -0,0 +1,149 @@
+                Assumes.Present(container);
+                IServiceBroker serviceBroker = container.GetFullAccessServiceBroker();
+                return new ServiceBrokerClient(serviceBroker, _joinableTaskFactory);
+            }, _joinableTaskFactory);
+        }
+
+        public override Task ActivateAsync()
+        {
+            // Nothing to do here in Visual Studio Online Environments mode. We do not create the pane
+            return Task.CompletedTask;
+        }
+
+        public override Task ClearAsync()
+        {
+            return CloseChannelAsync();
+        }
+
+        public override Task WriteAsync(string text)
+        {
+            return WriteToOutputChannelAsync(_channelId, _channelOutputName, text, CancellationToken.None);
+        }
+
+        private async Task WriteToOutputChannelAsync(string channelId, string displayNameResourceId, string content, CancellationToken cancellationToken)
+        {
+            using (await _pipeLock.EnterAsync())
+            {
+                if (_channelPipeWriter == null)
+                {
+                    var pipe = new Pipe();
+
+                    using (var outputChannelStore = await (await _serviceBrokerClient.GetValueAsync()).GetProxyAsync<IOutputChannelStore>(VisualStudioServices.VS2019_4.OutputChannelStore, cancellationToken))
+                    {
+                        if (outputChannelStore.Proxy != null)
+                        {
+                            await outputChannelStore.Proxy.CreateChannelAsync(channelId, displayNameResourceId, pipe.Reader, TextEncoding, cancellationToken);
+                            _channelPipeWriter = pipe.Writer;
+
+                            // write any deferred messages
+                            foreach (var s in _deferredOutputMessages)
+                            {
+                                // Flush when the original content is logged below
+                                await _channelPipeWriter.WriteAsync(GetBytes(content), cancellationToken);
+                            }
+                            _deferredOutputMessages.Clear();
+                        }
+                        else
+                        {
+                            // OutputChannel is not available so cache the output messages for later
+                            _deferredOutputMessages.Add(content);
+                            return;
+                        }
+                    }
+                }
+                await _channelPipeWriter.WriteAsync(GetBytes(content), cancellationToken);
+                await _channelPipeWriter.FlushAsync(cancellationToken);
+            }
+        }
+
+        private static byte[] GetBytes(string content)
+        {
+            return TextEncoding.GetBytes(content);
+        }
+
+        private async Task CloseChannelAsync()
+        {
+            using (await _pipeLock.EnterAsync())
```

Comment:
Is there any chance that may be data in the _deferredOutputMessages? Do we want these to be written?

Reply:
Not really. We only close the channel when we want to clear the output.
The way to clear the output is to close and reopen the channel.